### DISCUSSION TITLE
Is there a way to handle externally generated live pcap files ?

### DISCUSSION THREAD

**Discussion by Sraq-Zit** [nfstream/nfstream#98] [Q&A]
Hello,

Imagine we have a use case where pcap files are generated by another tool from an interface, and we want to process these pcap files using nfstream.
Processing file by file causes flows starting in one file and ending in another to generate two flows for each file if the pcap files are not merged (correct me if I'm wrong).
Another idea is to make the tool generate one file with a reasonable size and keep updating it with new packets and overwriting old ones when the size is exceeded, in this case, can nfstream read the pcap file in a streaming mode ?

Any other ideas ?

Thank you

**Comment by drnpkr** [comment]
Hi,

It completely makes sense. This problem indeed occurs when processing traces divided into separate pcap files.

An option you have not mentioned but could serve as a workaround is to merge the pcaps (e.g., via the mergecap tool). Sure, only if its final size will be of reasonable size. 

A couple of months ago I have started implementing a new functionality in NFStream. Specifically, an option to enter a range of pcap files as input, and process them sequentially without flushing the flows in the flow cache when moving to a new pcap. However, I have been kept back due to family matters.

Do you want to give it a try and implement it yourself? Sadly, I won't be able to get back to my fork anytime soon.

**Comment by Sraq-Zit** [comment]
Hi,

Thank you for your quick response.
Can you share the fork ?

**Reply by drnpkr** [reply]
Sure. I will need to check how far did I get, so I can give you a proper description. Do you mind reaching out to me via e-mail? adrian[dot]pekar[at]gmail[dot]com
